map.c

```diff
@@ -101,8 +101,21 @@ map_resize(map_t *m, int new_cap)
     if (!nm) {
         return -1;
     }
-    memcpy(m, nm, sizeof(map_t));
-    map_free(nm);
+
+    for (int i = 0; i < m->len; i++) {
```

Comment:
Ugh... This is what I initially had (for the most part)(this impl is better) and I decided to use memcpy for some reason.